### PR TITLE
First Version Of Encumbrance By Sentence

### PR DESCRIPTION
**Pull request by bombasticSlacks**
#### Summary
None

#### Purpose of change
@Aerin-of-the-Toast mentioned that they wanted to overhaul all the helmets in the game since they found research that the major thing that effects helmet encumbrance is weight. They created a chart with some brackets based on weight and them some flat modifiers so that they could fairly modify all helmets... This sounds like part of what I want to do in the future to **all encumbrance values** so I figured I could build some preliminary infrastructure now so helmets should be overhauled in the desired uniform way while leading to further work in the future.
 

#### Describe the solution

Limbs can now describe a set of fixed weight to encumbrance values for the limb in json. These can then be used as reference points to interpolate other weight to encumbrance values.

When defining an armor entry instead of giving a number for encumbrance you can give a set of encumberance_modifiers. 

This will make it so encumbrance is calculated by doing a linear interpolation on the two closest weight to encumbrance values for the limb and then modifying it by any encumbrance_modifiers that have been provided for the item.

The encumbrance modifiers for helmets are:

- IMBALANCED +10
- RESTRICTS_NECK +10
- WELL_SUPPORTED -10
- NONE +0

Otherwise everything just works ™️  still. 

#### Testing
image
it works for the army helmet and motorcycle helmet I added in this PR.

#### Additional context
I'm starting this as draft for now so I can get some feedback and write some documentation tomorrow.

### CODE REVIEW

**Comment by bombasticSlacks** [comment]
Note for me for tomorrow make sure a negative number isn't possible.

File: src/item_factory.cpp [hunk 1]
```diff
@@ -2657,7 +2685,10 @@ void armor_portion_data::deserialize( const JsonObject &jo )
 
     optional( jo, false, "rigid_layer_only", rigid_layer_only, false );
 
-    if( jo.has_array( "encumbrance" ) ) {
+    if( jo.has_array( "encumberance_modifiers" ) ) {
```
Comment (anothersimulacrum) [inline src/item_factory.cpp:2688]: You'll need to do this everywhere else too.
```suggestion
    if( jo.has_array( "encumbrance_modifiers" ) ) {
```

**Comment by omni-Var** [comment]
Are these values offset by strength scores? The football player obviously has an easier time fielding this encumbering weight than the office nerd that never works out.

**Comment by bombasticSlacks** [comment]
the first implementation will just be for helmets, which doesn't have much to do with strength. In the future when the system is expanded, I'm not sure.

**Pull request merged**